lessons/de/chapter_5.yaml

```
@@ -0,0 +1,224 @@
+- title: Das Löschen ist hierarchisch
+  content_markdown: >
+    Wenn eine Struktur gelöscht wird, wird zuerst die Struktur selbst und dann die Struktur selbst gelöscht
+    Kinder werden einzeln abgesetzt und so weiter.
+
+
+    Speicherdetails:
+
+    * Durch die automatische Freigabe von Speicher trägt Rust dazu bei, dass es weniger Speicher gibt
+    Speicherlecks.
+
+    * Speicherressourcen können nur einmal gelöscht werden.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=struct%20Bar%20%7B%0A%20%20%20%20x%3A%20i32%2C%0A%7D%0A%0Astruct%20Foo%20%7B%0A%20%20%20%20bar%3A%20Bar%2C%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20foo%20%3D%20Foo%20%7B%20bar%3A%20Bar%20%7B%20x%3A%2042%20%7D%20%7D%3B%0A%20%20%20%20println!(%22%7B%7D%22%2C%20foo.bar.x)%3B%0A%20%20%20%20%2F%2F%20se%20renun%C8%9B%C4%83%20la%20foo%20mai%20%C3%AEnt%C3%A2i%0A%20%20%20%20%2F%2F%20apoi%20se%20renun%C8%9B%C4%83%20la%20foo.bar%0A%7D%0A
+- title: Eigentumsübertragung
+  content_markdown: >
+    Wenn ein Eigentümer als Argument an eine Funktion übergeben wird, wird der Eigentümer dorthin verschoben
+    der Funktionsparameter.
+
+    Nach einer **Verschiebung** kann die Variable in der ursprünglichen Funktion nicht mehr vorhanden sein
+    gebraucht.
+
+
+    Speicherdetails:
+
+    * Während einer **Verschiebung** wird der Stapelspeicher des Eigentümerwerts in den kopiert
+    Parameterstapelspeicher des Funktionsaufrufs.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=struct%20Foo%20%7B%0A%20%20%20%20x%3A%20i32%2C%0A%7D%0A%0Afn%20do_something(f%3A%20Foo)%20%7B%0A%20%20%20%20println!(%22%7B%7D%22%2C%20f.x)%3B%0A%20%20%20%20%2F%2F%20se%20renun%C8%9B%C4%83%20la%20f%20aici%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20foo%20%3D%20Foo%20%7B%20x%3A%2042%20%7D%3B%0A%20%20%20%20%2F%2F%20foo%20este%20cedat%20func%C8%9Biei%20do_something%0A%20%20%20%20do_something(foo)%3B%0A%20%20%20%20%2F%2F%20foo%20nu%20mai%20poate%20fi%20folosit%0A%7D%0A
+- title: Rückgabe des Eigentums
+  content_markdown: |
+    Der Besitz kann auch von einer Funktion zurückgegeben werden.
+  code: >-
+    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=struct%20Foo%20%7B%0A%20%20%20%20x%3A%20i32%2C%0A%7D%0A%0Afn%20do_something()%20-%3E%20Foo%20%7B%0A%20%20%20%20Foo%20%7B%20x%3A%2042%20%7D%0A%20%20%20%20%2F%2F%20proprietatea%20este%20cedat%C4%83%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20foo%20%3D%20do_something()%3B%0A%20%20%20%20%2F%2F%20foo%20devine%20proprietar%0A%20%20%20%20%2F%2F%20se%20renun%C8%9B%C4%83%20la%20foo%20din%20cauza%20termin%C4%83rii%20func%C8%9Biei%0A%7D%0A
+- title: Eigentumsleihe mit Referenzen
+  content_markdown: |
+    Referenzen ermöglichen uns den Zugriff auf eine Ressource mit dem Operator '&'.
```

Use backticks for inline code:
```suggestion
    Referenzen ermöglichen uns den Zugriff auf eine Ressource mit dem Operator `&`.
```